buildtest/modules/util.py

```diff
@@ -1,48 +1,156 @@
-import subprocess
+from buildtest.tools.system import BuildTestCommand
 
 
 def get_all_collections():
-    """Return all module collection """
+    """Get all Lmod user collections that is retrieved by running ``module -t savelist``.
+
+     :return: Return all module collections
+     :rtype: list
+     """
+
     collections = "module -t savelist"
-    ret = subprocess.Popen(
-        collections, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-    )
-    out = ret.communicate()[1]
-    tmp = out.decode("utf-8").split()
+    cmd = BuildTestCommand()
```

Comment:
I'm concerned that here we have shell set to True, which I think is considered bad practice:

```bash
        self.ret = subprocess.Popen(
            cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
        )
```
I think there are other ways to handle this (e.g., for sregistry I use Popen, see [here](https://github.com/singularityhub/sregistry-cli/blob/60e2e48d0fbf650f459d99f7b1d28f0c1e44069d/sregistry/utils/terminal.py#L126). Overall, having a consistent function to execute commands is great - but I don't think we want to use shell=True. I'll open an issue to refactor this for a future PR.

Reply:
https://github.com/HPC-buildtest/buildtest-framework/issues/182